yazi-scheduler/src/file/file.rs

```diff
@@ -81,16 +81,25 @@ impl File {
 				};
 
 				ok_or_not_found(fs::remove_file(&task.to).await)?;
-				#[cfg(unix)]
-				{
-					fs::symlink(src, &task.to).await?
-				}
-				#[cfg(windows)]
-				{
-					if meta.is_dir() {
-						fs::symlink_dir(src, &task.to).await?
+
+				if task.hard {
+					if !src.is_symlink() && src.is_dir() {
```

Comment:
I added `!src.is_symlink()` here to cover the special case where the yanked file is a symlink to a directory. With this version, a hardlink of the symlink is created instead of following the original symlink and hardlinking the destination.

Regular directories and symlinks to non-directory files were working correctly even before this change.